knowledge-base/detect-row-detail-expansion-collapse-maui-datagrid.md

```diff
@@ -0,0 +1,229 @@
+    {
+        var items = new ObservableCollection<Data>
+        {
+            new Data { Country = "India", Capital = "New Delhi" , Details = "New Delhi is the capital of India and a part of the National Capital Territory of Delhi (NCT). New Delhi is the seat of all three branches of the Government of India, hosting the Rashtrapati Bhavan, Sansad Bhavan, and the Supreme Court."},
+            new Data { Country = "South Africa", Capital = "Cape Town", Details = "Cape Town is South Africa's oldest city. It serves as the country's legislative capital, being the seat of the South African Parliament.It is the country's second-largest city (after Johannesburg) and the largest in the Western Cape."},
+            new Data { Country = "Nigeria", Capital = "Abuja" , Details = "Abuja is the capital city of Nigeria. When it was decided to move the national capital from Lagos in 1976, a capital territory was chosen for its location near the centre of the country. The planned city is located in the centre of what is now the Federal Capital Territory." },
+            new Data { Country = "Singapore", Capital = "Singapore" , Details = "Singapore is the capital city of the Republic of Singapore. It occupies the southern part of Singapore Island. Its strategic position on the strait between the Indian Ocean and South China Sea, complemented by its deepwater harbour, has made it the largest port in Southeast Asia." }
+        };
+
+        this.Items = items;
+
+        this.ExpandedItems = new ObservableCollection<Data>();
+
+        // Approach 1
+        this.ExpandedItems.CollectionChanged += ExpandedItems_CollectionChanged;
+    }
+
+    // Approach 1
+    private void ExpandedItems_CollectionChanged(object? sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+    {
+        if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Add)
+        {
+            // your logic here
+        }
+        else if(e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Remove)
+        {
+            // your logic here
+        }
+    }
+
+    public ObservableCollection<Data> Items { get; set; }
+
+    public ObservableCollection<Data> ExpandedItems { get; set; }
+}
+
+public class Data
+{
+    public string Country { get; set; }
+    public string Capital { get; set; }
+    public string Details { get; set; }
+}
+```
+
+
+### Approach 2: Using the `ToggleRowDetailsButtonTap` Command
+
+**1.** Utilize the `ToggleRowDetailsButtonTap` command provided by the DataGrid. This command is triggered whenever the toggle button for row details is tapped. In the command's execution logic, add a flag or a mechanism to check whether the item related to the toggled row detail is present in the `ExpandedRowDetails` collection. An item's presence indicates an expansion, while its absence indicates a collapse.
+
+Implement the necessary logic based on this condition to handle the expansion or collapse state.
```

Comment:
Is this step 2? Or does it belong to step 1?

Reply:
step 2